src/Formats/JSONEachRowUtils.cpp

```diff
@@ -92,6 +100,167 @@ static std::pair<bool, size_t> fileSegmentationEngineJSONEachRowImpl(ReadBuffer
     return {loadAtPosition(in, memory, pos), number_of_rows};
 }
 
+template <const char opening_bracket, const char closing_bracket>
+static String readJSONEachRowLineIntoStringImpl(ReadBuffer & in)
```

Comment:
This looks like a copy paste of segmentation function used for parallel parsing

Reply:
Yep. To be honest, don't remember why I couldn't generalize it (maybe just forgot to do it), will try now.